test/src/main/java/io/strimzi/test/k8s/KubeClient.java

```diff
@@ -663,8 +672,19 @@ public List<ClusterRole> listClusterRoles() {
         return client.rbac().clusterRoles().list().getItems();
     }
 
-    public void createOrReplaceClusterRoles(ClusterRole clusterRole) {
-        client.rbac().clusterRoles().resource(clusterRole).createOrReplace();
+    /**
+     * Method for creating the specified ClusterRole.
+     * In case that the ClusterRole is already created, it is being updated.
+     * This can be caused by not cleared ClusterRoles from other tests on in case we shut down the test before the cleanup
+     * phase. It should not have an impact on the functionality, we just update the ClusterRole.
+     * @param clusterRole ClusterRole that we want to create or update
+     */
+    public void createOrUpdateClusterRoles(ClusterRole clusterRole) {
+        try {
+            client.rbac().clusterRoles().resource(clusterRole).create();
+        } catch (KubernetesClientException e) {
+            client.rbac().clusterRoles().resource(clusterRole).update();
+        }
```

Comment:
Same as above. You should check for the right error here. What are the reasons for SA not being cleaned up?

There seem to be a lot of these. I won't repeat this comment anymore.